functions/public/onload-expression.js

```diff
@@ -2,17 +2,29 @@
   window._pagarmeHash = function (card) {
     return new Promise(function (resolve, reject) {
       // https://github.com/pagarme/pagarme-js#using-encryption-key
+      const usedCard = { 
+        card: {
+          card_number: card.number,
+          card_holder_name: card.name,
+          card_expiration_date: card.month.toString() + card.year.toString(),
+          card_cvv: card.cvc
+        }
+      }
+      const validateObject = window.pagarme.validate(usedCard)
+      const objectCardValidated = validateObject && validateObject.card
+      for (let key in objectCardValidated) {
+        if (Object.hasOwnProperty.call(objectCardValidated, key)) {
+          if (!objectCardValidated[key]) {
+            return reject()
+          }
+        }
+      }
```

Comment:
Complicou demais aqui não? 😄 

Algum motivo pra não deixar só:

```suggestion
      if (!objectCardValidated.card) {
        return reject(new Error('...'))
      }
```

?

Reply:
Sim, porque nosso validador por exemplo, se você coloca uma data de vencimento que já passou, ele considera como verdadeira, esse checa quaisquer informações do card, deixando uma análise mais completa

Reply:
An? Acho que você não entendeu não haha

Reply:
Eu não mexi no validator aí, seguiu validando todos os fields no objeto `card`, incluindo a data...
Só tirei esse `for in` que acho que você pegou de algum exemplo pronto aí com mais campos envolvidos, pela ref deles https://pagarme.github.io/pagarme-js/validations_index.js.html `body.cnpj` por exemplo, não é o caso aqui

Reply:
> Eu fui ver a ref deles justamente porque quando bati o olho nesse loop achei confuso, me parece que a implementação aqui poderia ser bem mais simples

Reply:
Ahh ele não valida o `card` inteiro

Reply:
Que gambiarrinha haha

Reply:
> Eu não mexi no validator aí, seguiu validando todos os fields no objeto `card`, incluindo a data... Só tirei esse `for in` que acho que você pegou de algum exemplo pronto aí com mais campos envolvidos, pela ref deles https://pagarme.github.io/pagarme-js/validations_index.js.html `body.cnpj` por exemplo, não é o caso aqui

Peguei não kkk. Ele retorna a validação por propriedade, por isso checando individualmente cada propriedade